### PR TITLE
feat(account): setup account premium model, middlewares

### PR DESCRIPTION
**Pull request by huytran46**
LOZ-23
- setup account premium model
- implement premium-only middleware
- apply for document/share endpoint
- apply for document/create endpoint. FREE only 1 document

### CODE REVIEW

**Comment by vudoan1708-cyber** [comment]
@tranqhuy46 Huy, check share okay roi, nh ma ong thieu check cho so luong document dc create nua.

PREMIUM: unlimited number of documents created && document sharing
NON-PREMIUM / FREE: 1 document created && no sharing

File: backend/src/shared/error/locale.en.ts [hunk 1]
```diff
@@ -17,6 +17,7 @@ const ERROR_EN: Record<ErrorCode, string> = {
   PAE013: "You cannot view this document",
   PAE014: "Wrong document's passcode",
   PAE015: "Cannot unshare (Wrong document's passcode)",
+  PAE0077: "Not a premium user",
```
Comment (vudoan1708-cyber) [inline backend/src/shared/error/locale.en.ts]: @tranqhuy46 K nghiêm trọng lắm, nh mà ông update cái error mes thành vầy giúp tui vs:
"You are using a free account.\n
Please upgrade your account for more premium features"

Comment (huytran46) [reply backend/src/shared/error/locale.en.ts]: roi nha Vu

Comment (vudoan1708-cyber) [reply backend/src/shared/error/locale.en.ts]: merge thẳng tiến

**Pull request merged**